### PR TITLE
Updated Haier integration documentation to support alarm triggers

### PR DESCRIPTION
**Pull request by paveldn**
## Description:

Updated documentation to support new features (alarm triggers and AWAY preset)

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#5965

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Review by jesserockz — DISMISSED** [review]

File: components/climate/haier.rst [hunk 1]
```diff
@@ -2,14 +2,14 @@ Haier Climate
...
 
 .. seo::
-    :description: Instructions for setting up a Haier climate devices.
+    :description: Instructions for setting up Haier climate devices.
     :image: air-conditioner.svg
 
 This is an implementation of the ESPHome component to control HVAC on the base of the SmartAir2 and hOn Haier protocols (AC that is controlled by the hOn or SmartAir2 application).
 
-There are two versions of the Haier protocol. The older one is using an application called SmartAir2 and the newer one - an application called hOn. Both protocols are compatible on the transport level but have different commands to control appliances.
+There are two versions of the Haier protocol. The older one is using is an application called SmartAir2 and the newer one - an application called hOn. Both protocols are compatible on the transport level but have different commands to control appliances.
```
Comment (kbx81) [inline components/climate/haier.rst]: The added "is" isn't necessary. 🤔

Comment (kbx81) [reply components/climate/haier.rst]: I might reword this as follows:

```suggestion
There are two versions of the Haier protocol: the older version uses an application called SmartAir2 while the newer version uses an application called hOn. Both protocols are compatible on the transport level but utilize different commands to control appliances.
```

File: components/climate/haier.rst [hunk 2]
```diff
@@ -2,14 +2,14 @@ Haier Climate
...
 
-Older Haier models controlled by the SmartAir2 application are using the KZW-W002 module. This module can’t be reused, and you need to replace it with an ESP (RPI pico w) module. The USB connector on a board doesn’t support the USB protocol. It is a UART port that just uses a USB connector. To connect the ESP board to your AC you can cut a USB type A cable and connect wires to the climate connector.
+Older Haier models controlled by the SmartAir2 application are using the KZW-W002 module. This module can’t be reused, and you need to replace it with an ESP (RPI pico w) module. The USB connector on the board doesn’t support the USB protocol. It is a UART port that just uses a USB connector. To connect the ESP board to your AC you can cut a USB type A cable and connect wires to the climate connector.
```
Comment (kbx81) [inline components/climate/haier.rst]:
```suggestion
Older Haier models controlled by the SmartAir2 application are using the KZW-W002 module. This module can’t be reused, and you need to replace it with an ESP or RPI Pico W module. The USB connector on the board doesn’t support the USB protocol. It is a UART port that just uses a USB connector. To connect the ESP board to your AC you can cut a USB type A cable and connect wires to the climate connector.
```

**Review by kbx81 — CHANGES_REQUESTED** [review]
Couple little nits, otherwise LGTM

**Comment by paveldn** [comment]
@kbx81 Thanks. English is not my native language, sometimes, I make stupid mistakes

**Review by kbx81 — APPROVED** [review]

**Comment by jesserockz** [comment]
> kbx81 Thanks. English is not my native language, sometimes, I make stupid mistakes

English **is** my native language and I make stupid mistakes :stuck_out_tongue_closed_eyes:
